### PR TITLE
Revert "Remove -Werror"

### PR DESCRIPTION
**Pull request by Doy-lee**
This reverts commit 707c2f836b3136bc88b5aed374a4b3128f9596a7.

### CODE REVIEW

**Comment by Doy-lee** [comment]
This is most likely ready to go. Can't verify on Mac OS because RandomX build error but all other platforms signalling ok.

@jagerman

**Comment by jagerman** [comment]
I have mixed feelings about this PR.

On the one hand I think we definitely *should* be building locally and on the travis-ci builds with `-Werror`.  I even think we should go further and not disable a bunch of warnings like is currently being done in CMakeLists.txt.  (For warnings that really are difficult to avoid there is always a code-localized [warning disabling pragma](https://gcc.gnu.org/onlinedocs/gcc/Diagnostic-Pragmas.html) for gcc/clang, and similar for MSVC, not that MSVC is affected by this).

But the reason that I don't like enabling it unconditionally is that it *does* break the compilation when compiler versions get upgraded with new warnings or new versions of boost or other libraries come out, and I'm assuming this is why Monero disabled it in the first place.

Case in point: current `dev` branch is not warning-free under gcc 9 (PR #840), and I'm pretty sure that has been the case of the code case in every new gcc release.

**Comment by jagerman** [comment]
Lokinet, for example, has a `WARNINGS_AS_ERRORS` [cmake option](https://github.com/loki-project/loki-network/blob/2f9c83e287f8d666d934af67d11866edf3603cf0/CMakeLists.txt#L30) that enables `-Werror` that defaults to off, but a build configured with `cmake -DWARNINGS_AS_ERRORS=ON ..` turns it on.

**Comment by Doy-lee** [comment]
Okay sounds, reasonable. I've added in similar fashion, -D WARNING_AS_ERRORS=ON for release stuff.

File: contrib/epee/tests/src/CMakeLists.txt [hunk 1]
```diff
@@ -14,8 +14,8 @@ IF (MSVC)
 	include_directories(SYSTEM platform/msvc)
 ELSE()
 	# set stuff for other systems
-	SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -std=c11 -Wall")
-	SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11 -Wall -Wno-reorder")
+	SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -std=c11 -Wall -Werror")
+	SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11 -Wall -Werror -Wno-reorder")
```
Comment (jagerman) [inline contrib/epee/tests/src/CMakeLists.txt]: Should this be `${WARNINGS_AS_ERRORS_FLAG}`?

Comment (Doy-lee) [reply contrib/epee/tests/src/CMakeLists.txt]: Added

**Pull request merged**